### PR TITLE
SONARPY-1319 Rule S6543: Type hints of generic types should have a type parameter

### PR DESCRIPTION
**Pull request by joke1196**

### CODE REVIEW

**Review by gyula-sallai-sonarsource — CHANGES_REQUESTED** [review]
Looks good to me, I've left only smaller comments on tests and code style.

File: python-checks/src/test/java/org/sonar/python/checks/GenericTypeWithoutArgumentCheckTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import org.junit.Test;
+import org.sonar.python.checks.utils.PythonCheckVerifier;
+
+import static org.junit.Assert.*;
```
[the remark this answers is not in the document]
Comment (guillaume-dequenne-sonarsource) [reply python-checks/src/test/java/org/sonar/python/checks/GenericTypeWithoutArgumentCheckTest.java:25]: (Jumping on this as I just see this: I think the code style in our developer box should prevent any wildcard import from being applied?)

File: python-checks/src/test/resources/checks/genericTypeWithoutArgument.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import typing
+
```
[the remark this answers is not in the document]
Comment (joke1196) [reply python-checks/src/test/resources/checks/genericTypeWithoutArgument.py:2]: Good catch this one could have failed on non-qualified expression.

**Review by gyula-sallai-sonarsource — APPROVED** [review]
Looks good to me, nice job!

**Pull request merged**